src/components/Scheme/ContributionRewardExtRewarders/rewardersProps.ts

```diff
@@ -0,0 +1,52 @@
+import { ISchemeState } from "@daostack/client";
+import { getArc } from "arc";
+
+export const hasRewarderContract = (schemeState: ISchemeState): boolean => {
+  return !!schemeState.contributionRewardExtParams && !!schemeState.contributionRewardExtParams.rewarder;
+};
+
+export const rewarderContractName = (schemeState: ISchemeState): string => {
+  if (hasRewarderContract(schemeState)) {
+    const contractInfo = getArc().getContractInfo(schemeState.contributionRewardExtParams.rewarder);
+    if (contractInfo) {
+      return contractInfo.name;
+    } else {
+      // eslint-disable-next-line no-console
+      console.error(`rewarder contract not found: ${schemeState.contributionRewardExtParams.rewarder}`);
+    }
+  }
+  return null;
+};
+
+export interface ICrxRewarderProps
+{
+  friendlyName: string;
+  contractName: string;
+  shortDescription: string;
+  // description: string;
+}
+
+/**
+ * If the scheme is a ContributionRewardExt, return props for its rewarder contract, if any
+ * @param scheme 
+ */
+export const getCrxRewarderProps = (scheme: ISchemeState): Promise<ICrxRewarderProps> | null  => {
+  const contractName = rewarderContractName(scheme);
+  // dynamic imports are enabled and optimized by @babel/plugin-syntax-dynamic-import
+  return contractName ?  import(`./${contractName}/props.json`) : null;
+};
+
+export enum CrxRewarderComponentType {
+  List,
+  Details,
+  CreateProposal
+}
+export const getCrxRewarderComponent = (schemeState: ISchemeState, type: CrxRewarderComponentType): Promise<any> => {
+  const contractName = rewarderContractName(schemeState);
+  if (contractName) {
+    return import(`components/Scheme/ContributionRewardExtRewarders/${contractName}/${CrxRewarderComponentType[type]}`)
```

Comment:
Doing dynamic imports is a Big Change, with consequences forr the bundling process. I would only do this if I had a Really Good reason... do we have such a reason?

Reply:
The really good reason is so we don't have to statically reference and load all rewarding contract components.  Instead we are lazy loading, which is efficient for memory usage and bandwidth.

I don't see this as a big change, wondering why you say this?  When doing dynamic imports, which we can (and I think likely should) do elsewhere when we import JSON for generic schemes, and which I also do elsewhere in the competition code for JSON, then we want them to be bundled smartly -- they aren't by default.  The plugin I am using manages this nicely.

Reply:
* The problem is that if you import modules in this way (using live code) it is much harder (actually impossible) to calculate the dependencies of the module that does the import, and so it is harder for webpack to split the bundle into coherent chunks (I guess, in fact I do not even really understand how webpack knows what to include)

*  We are curently bundling all our code together, and this import syntax does not change that.So I do not see how this this "lazy loading" or "efficient for memory usage and bandwidth?", and also I do not see how this would help us move towards that  (on the contrary, see above)

any case, this is not blocking, but we may want to change this when we do bunde optimizing...

Reply:
dynamic importing is a good idea where possible i think. it makes for smaller bundles up front, though some small amount of waiting later on for the user...